test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -36,7 +36,28 @@ def semantic_version(v):
     return v
 
 
-def version(for_collection=False):
+TAGGED_VERSION_RE = re.compile('^([^.]+.[^.]+):(.*)$')
+
+
+def tagged_version(v):
+    if not isinstance(v, string_types):
+        # Should never happen to versions tagged by code
+        raise Invalid('Tagged version must be a string')
+    m = TAGGED_VERSION_RE.match(v)
+    if not m:
+        # Should never happen to versions tagged by code
+        raise Invalid('Tagged version does not match format')
+    collection = m.group(1)
+    version = m.group(2)
+    if collection != 'ansible.builtin':
+        semantic_version(version)
```

Comment:
Hmmm... if semantic_versions are followed, new features should always be inrroduced at  x.y.0, right?  I'm not sure if we want to enforce that, though.

(And I guess 0.y semantic versions are one wrinkle in that)

If we had a schema on docs parse, we could use it to allow "x.y" on parse but turn it into "x.y.0" before validation.  Unfortunately, that doesn't help with a schema that's only used on test.

Reply:
I don't mind if we enforce this. I guess deprecations should also only happen in x.y.0 versions (and removals in x.0.0 versions).

Maybe these should be optional errors though (i.e. only returned as errors when `--enable-optional-errors` is passed to ansible-test).